### ISSUE TITLE
Opensips 2.2.4 crashes 

### ISSUE THREAD

**Issue by ferrored** [OpenSIPS/opensips#1132]
I'm using opensips installed on Ubuntu 16.04.2 LTS.
version: opensips 2.2.4 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll_lt, epoll_et, sigio_rt, select.
git revision: 1f3ae25cc
main.c compiled on 08:01:09 May 22 2017 with gcc 5.4.0

At the end of an outbound call to an UAC registered in a different sip domain , I get opensips v 2.2.4 crashes.

[output.txt](https://github.com/OpenSIPS/opensips/files/1031299/output.txt)

**Comment by bogdan-iancu** [comment]
@ferrored , what version of libfreeradius do you use ? also, does this crash happens all the time when pushing an CDR via RADIUS ? or is random ?

**Comment by ferrored** [comment]
ibfreeradius-client-dev         1.1.6-7   amd64        Enhanced RADIUS client library development files
libfreeradius-client2              1.1.6-7   amd64       Enhanced RADIUS client library

It seems it happens all the time

**Comment by ferrored** [comment]
yes, it was enabled but I noticed that when opensips crashed, RADIUS was unreachable.

**Comment by bogdan-iancu** [comment]
so, this crash occurred only was the RADIUS server was unreachable ?

I'm asking as the backtrace show the crash deep in the libfreeradius-client module, and it actually may be an issue with the lib, not with OpenSIPS.

**Comment by ferrored** [comment]
I confirm that the crash happens only when server is unreachable. I will test the new version of libfreeradius-client when it will be available in ubuntu repository

**Comment by bogdan-iancu** [comment]
Do you use the async support for RADIUS (the extra patch we provide) ? or is it stock radiusclient lib ?

**Comment by ferrored** [comment]
No, I'm not using the extra patch provided for the async support for RADIUS

**Comment by ferrored** [comment]
Should I?

**Comment by razvancrainea** [comment]
I have met the same situation as here in a different environment. Thanks to Brian Chamberlain from IP Tel, I managed to do further debugging and find the real issue.

It seems like the problem is only present in freeradius 1.1.6, and has been fixed in 1.1.7 release.

The workaround I found now is to specify the secret used by the acctservers directly in the /etc/radiusclient/radiusclient.conf file:

```
acctserver 127.0.0.1:1813:PASSWD
```

If you are not using a password, you can have something like this:
```
acctserver 127.0.0.1:1813:
```

Hopefully this sorts out your issue too. If it doesn't, try to upgrade your library to 1.1.7.
I will now close this ticket, since there is nothing we can do to fix it in OpenSIPS.
